client/forms/canvas.cpp

```diff
@@ -267,6 +286,60 @@ void GameLoopCanvas::OnMouseUpdate(wxMouseEvent& event)
   }
 }
 
+Item GameLoopCanvas::getMineableVoxel() {
+
+	glm::vec3 inFront = position;
+	BlockType block;
+	for (float i = 0.0f; i < 2.0f; i += 0.1f) {
+		inFront = position + lookat * i;
+		block = (*chunk_manager).get(inFront.x, inFront.y, inFront.z);
+		if (block != BlockType::Inactive) {
+			break;
+		}
+	}
+
+	if (block != BlockType::Inactive) {
+
+		switch (block) {
+		case BlockType::Ground:
+			return Item("Ground");
+		case BlockType::Water:
+			return Item("Water");
+		case BlockType::Sand:
+			return Item("Sand");
+		case BlockType::Wood:
+			return Item("Wood");
+		case BlockType::Flowers:
+			return Item("Flowers");
+		case BlockType::Ruby:
+			return Item("Ruby");
+		case BlockType::Leaves:
+			return Item("Leaves");
+		case BlockType::Stone:
+			return Item("Stone");
+		case BlockType::Grass:
+			return Item("Grass");
+		case BlockType::Brick:
+			return Item("Brick");
+		case BlockType::Party:
+			return Item("Party");
+		case BlockType::Coal:
+			return Item("Coal");
+		case BlockType::Iron:
+			return Item("Iron");
+		case BlockType::Gravel:
+			return Item("Gravel");
+		case BlockType::Diamond:
+			return Item("Diamond");
+		case BlockType::Gold:
+			return Item("Gold");
+		}
+
+		(*chunk_manager).set(inFront, BlockType::Inactive);
+	}
+	return NULL;
```

Comment:
Does this compile? How can you assign a struct to null? Did you mean to return a pointer here?